libraries/rush-lib/src/logic/base/BaseInstallManagerTypes.ts

```diff
@@ -84,8 +84,18 @@ export interface IInstallManagerOptions {
   /**
    * Filters to be passed to PNPM during installation, if applicable.
    * These restrict the scope of a workspace installation.
+   *
+   * @remarks
+   * Note that PNPM may arbitrarily ignore `--filter` (producing an unfiltered install) in certain situations,
+   * for example when `config.dedupe-peer-dependents=true` with PNPM 8.  Rush tries to circumvent this, under the
+   * assumption that a user who invokes a filtered install cares more about lockfile stability than duplication.
+   */
+  pnpmFilterArguments: string[];
+
+  /**
+   * The set of projects for which installation should be performed.
    */
-  filteredProjects: RushConfigurationProject[];
+  selectedProjects: Set<RushConfigurationProject>;
```

Comment:
Is `selectedProjects` actually needed any more? From what I can tell, it is only ever used by this code:

https://github.com/microsoft/rushstack/blob/236a0e52e6e0c79b687ed13d8cf5f163daed98cc/libraries/rush-lib/src/logic/base/BaseInstallManager.ts#L167-L170

...which just constructs a set of installed project names for `last-install.flag` to track.  But isn't that exactly the same list as `pnpmFilterArguments`?

Reply:
No, we aren't generating `--filter` args for every project anymore when doing a filtered install outside of subspace unless someone says `rush install --only <something>`. For the `--to` case, for example, we're generating args that use the pnpm filter syntax: https://pnpm.io/filtering